### PR TITLE
Local units delete modal

### PR DESCRIPTION
**Pull request by shreeyash07**
## Addresses:
- #1497 

## Changes
- Add modal for localunits deletion in localunit table and localunit form
- Update delete localunits api 

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsTable/LocalUnitTableActions/index.tsx [hunk 1]
```diff
@@ -76,6 +86,13 @@ function LocalUnitsTableActions(props: Props) {
         [setShowLocalUnitModalTrue],
     );
 
+    const handleDeleteLocalUnitClick = useCallback(
+        () => {
+            setShowDeleteLocalUnitModalTrue();
+        },
+        [setShowDeleteLocalUnitModalTrue],
+    );
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsTable/LocalUnitTableActions/index.tsx]: We don't need to define this function. We can directly pass `setShowDeleteLocalUnitModalTrue` where `hanldeDeleteLocalUnitClick` is used

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsTable/LocalUnitTableActions/index.tsx [hunk 2]
```diff
@@ -35,10 +37,12 @@ function LocalUnitsTableActions(props: Props) {
         localUnitName,
         localUnitId,
         isValidated,
-        onActionSuccess,
+        onValidationActionSuccess,
+        onDeleteActionSuccess,
     } = props;
 
     const { isCountryAdmin, isSuperUser } = usePermissions();
+    const { isAuthenticated: handleDeletePermission } = useAuth();
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsTable/LocalUnitTableActions/index.tsx]: This rename is not correct.

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsFormModal/LocalUnitsForm/index.tsx [hunk 3]
```diff
@@ -299,6 +309,13 @@ function LocalUnitsForm(props: Props) {
         [validate, localUnitId, setError, updateLocalUnit, addLocalUnit],
     );
 
+    const handleDeleteLocalUnitClick = useCallback(
+        () => {
+            setShowDeleteLocalUnitModalTrue();
+        },
+        [setShowDeleteLocalUnitModalTrue],
+    );
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsFormModal/LocalUnitsForm/index.tsx]: We don't need to define this function. We can just use setShowDeleteLocalUnitModalTrue

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        setError,
+        validate,
+    } = useForm(schema, { value: defaultFormValue });
+
+    const error = useMemo(
+        () => getErrorObject(formError),
+        [formError],
+    );
+
+    const { isAuthenticated: hasDeletePermission } = useAuth();
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx]: This rename is not correct.

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            requiredValidation: requiredStringCondition,
+        },
+    }),
+};
+
+interface Props {
+    localUnitId: number;
+    onDeleteActionSuccess?: () => void;
+    onClose: () => void;
+    localUnitName: string | null | undefined;
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx]: Shouldn't this always be defined when we are deleting a local unit? It should always have a name.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    } = useForm(schema, { value: defaultFormValue });
+
+    const error = useMemo(
+        () => getErrorObject(formError),
+        [formError],
+    );
+
+    const { isAuthenticated } = useAuth();
+
+    const hasDeletePermission = isAuthenticated;
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx]: This is still and issue. The way it was renamed was not the isssue.
The issue is regarding the logic.
Do we not have permissions defined on the server on what a user can perform?
Not all authenticated user should have delete permission.
example: we have guest users who can login but cannot perform any operation.

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsTable/LocalUnitTableActions/index.tsx [hunk 7]
```diff
@@ -35,30 +37,39 @@ function LocalUnitsTableActions(props: Props) {
...
+        onDeleteActionSuccess,
     } = props;
 
     const { isCountryAdmin, isSuperUser } = usePermissions();
+    const { isAuthenticated } = useAuth();
     const strings = useTranslation(i18n);
 
     const hasValidatePermission = isSuperUser || isCountryAdmin(countryId);
 
+    const hasDeletePermission = isAuthenticated;
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsTable/LocalUnitTableActions/index.tsx]: This is still and issue. The way it was renamed was not the isssue.
The issue is regarding the logic.
Do we not have permissions defined on the server on what a user can perform?
Not all authenticated user should have delete permission.
example: we have guest users who can login but cannot perform any operation.

File: app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+            deleteLocalUnit(formValues as LocalUnitDeprecateBody);
+        },
+        [deleteLocalUnit],
+    );
+
+    return (
+        <Modal
+            heading={resolveToString(
+                strings.deleteLocalUnitHeading,
+                { localUnitName: localUnitName ?? '' },
```
Comment (tnagorra) [inline app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx]: We can remove this fallback value as localUnitName is always string

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by samshara — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**